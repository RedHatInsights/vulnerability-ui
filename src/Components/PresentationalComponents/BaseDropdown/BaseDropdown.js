import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import {
    Dropdown,
    DropdownToggle
} from '@patternfly/react-core';
import { CaretDownIcon } from '@patternfly/react-icons';
import messages from '../../../Messages';

const BaseDropdown = ({ dropdownItems, disabled, ...props }) => {
    const [isOpen, setIsOpen] = useState(false);
    const intl = useIntl();

    return (
        <Dropdown
            toggle={
                <DropdownToggle
                    data-testid="dropdown-toggle"
                    onToggle={() => setIsOpen(!isOpen)}
                    toggleIndicator={CaretDownIcon}
                    isDisabled={disabled}
                    className="dropdown-toggle"
                >
                    {intl.formatMessage(messages.actions)}
                </DropdownToggle>
            }
            isOpen={isOpen}
            dropdownItems={dropdownItems}
            {...props}
        />
    );
};

BaseDropdown.propTypes = {
    dropdownItems: PropTypes.array,
    disabled: PropTypes.bool
};

export default BaseDropdown;

