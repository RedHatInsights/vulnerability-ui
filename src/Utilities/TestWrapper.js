import propTypes from 'prop-types';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

const TestWrapper = ({ children, store }) => {
    return (
        <IntlProvider locale="en">
            { store ?
                <Provider store={store}>
                    <Router>
                        {children}
                    </Router>
                </Provider> : children
            }
        </IntlProvider>
    );
};

TestWrapper.propTypes = {
    children: propTypes.node,
    store: propTypes.object
};

export default TestWrapper;
