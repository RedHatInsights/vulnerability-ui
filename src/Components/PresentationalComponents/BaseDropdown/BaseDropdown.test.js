import BaseDropdown from './BaseDropdown';
import { DropdownItem } from '@patternfly/react-core';
import { render, screen, waitFor } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import userEvent from '@testing-library/user-event';
import TestWrapper from '../../../Utilities/TestWrapper';

const mockStore = configureStore([store => next => action => {}]);
let store = mockStore({ });

const item1_action = jest.fn();

describe('BaseDropdown component', () => {
    it('Should match snapshot', () => {
        const items = [
            <DropdownItem key="item1" component="button" onClick={() => item1_action()}>
                Item 1
            </DropdownItem>
        ];

        const { asFragment } = render(
            <TestWrapper store={ store }>
                <BaseDropdown dropdownItems={items} />
            </TestWrapper>
        );

        expect(asFragment()).toMatchSnapshot();
    })

    it('Should be disabled when disabled prop is passed', () => {
        const items = [
            <DropdownItem key="item1" component="button" onClick={() => item1_action()}>
                Item 1
            </DropdownItem>
        ];

        const { asFragment } = render(
            <TestWrapper store={ store }>
                <BaseDropdown dropdownItems={ items } disabled={ true } />
            </TestWrapper>
        );

        expect(asFragment()).toMatchSnapshot();
    })

    it('Should call the onClick handler when clicked', async () => {
        const items = [
            <DropdownItem data-testid="dropdown-item1" key="item1" component="button" onClick={() => item1_action()}>
                Item 1
            </DropdownItem>
        ]

        render(
            <TestWrapper store={ store }>
                <BaseDropdown dropdownItems={items} />
            </TestWrapper>
        );

        userEvent.click(screen.getAllByTestId('dropdown-toggle')[0]); // open dropdown
        await waitFor(() => userEvent.click(screen.getByTestId('dropdown-item1'))); // click first item

        expect(item1_action).toHaveBeenCalled();
    })
});
