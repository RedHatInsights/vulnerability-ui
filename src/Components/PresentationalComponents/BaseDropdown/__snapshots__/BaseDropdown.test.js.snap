// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDropdown component Should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <BaseDropdown
        dropdownItems={
          Array [
            <DropdownItem
              component="button"
              onClick={[Function]}
            >
              Item 1
            </DropdownItem>,
          ]
        }
      >
        <Dropdown
          dropdownItems={
            Array [
              <DropdownItem
                component="button"
                onClick={[Function]}
              >
                Item 1
              </DropdownItem>,
            ]
          }
          isOpen={false}
          toggle={
            <DropdownToggle
              className="dropdown-toggle"
              onToggle={[Function]}
              toggleIndicator={[Function]}
            >
              Actions
            </DropdownToggle>
          }
        >
          <DropdownWithContext
            autoFocus={true}
            className=""
            direction="down"
            dropdownItems={
              Array [
                <DropdownItem
                  component="button"
                  onClick={[Function]}
                >
                  Item 1
                </DropdownItem>,
              ]
            }
            isGrouped={false}
            isOpen={false}
            isPlain={false}
            isText={false}
            menuAppendTo="inline"
            onSelect={[Function]}
            position="left"
            toggle={
              <DropdownToggle
                className="dropdown-toggle"
                onToggle={[Function]}
                toggleIndicator={[Function]}
              >
                Actions
              </DropdownToggle>
            }
          >
            <div
              className="pf-c-dropdown"
              data-ouia-component-id="OUIA-Generated-Dropdown-1"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <DropdownToggle
                aria-haspopup={true}
                className="dropdown-toggle"
                getMenuRef={[Function]}
                id="pf-dropdown-toggle-id-1"
                isOpen={false}
                isPlain={false}
                isText={false}
                key=".0"
                onEnter={[Function]}
                onToggle={[Function]}
                parentRef={
                  Object {
                    "current": <div
                      class="pf-c-dropdown"
                      data-ouia-component-id="OUIA-Generated-Dropdown-1"
                      data-ouia-component-type="PF4/Dropdown"
                      data-ouia-safe="true"
                    >
                      <button
                        aria-expanded="false"
                        aria-haspopup="true"
                        class="pf-c-dropdown__toggle dropdown-toggle"
                        data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                        data-ouia-component-type="PF4/DropdownToggle"
                        data-ouia-safe="true"
                        id="pf-dropdown-toggle-id-1"
                        type="button"
                      >
                        <span
                          class="pf-c-dropdown__toggle-text"
                        >
                          Actions
                        </span>
                        <span
                          class="pf-c-dropdown__toggle-icon"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 320 512"
                            width="1em"
                          >
                            <path
                              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>,
                  }
                }
                toggleIndicator={[Function]}
              >
                <Toggle
                  aria-haspopup={true}
                  bubbleEvent={false}
                  className="dropdown-toggle"
                  data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                  data-ouia-component-type="PF4/DropdownToggle"
                  data-ouia-safe={true}
                  getMenuRef={[Function]}
                  id="pf-dropdown-toggle-id-1"
                  isActive={false}
                  isDisabled={false}
                  isOpen={false}
                  isPlain={false}
                  isPrimary={false}
                  isSplitButton={false}
                  isText={false}
                  onEnter={[Function]}
                  onToggle={[Function]}
                  parentRef={
                    Object {
                      "current": <div
                        class="pf-c-dropdown"
                        data-ouia-component-id="OUIA-Generated-Dropdown-1"
                        data-ouia-component-type="PF4/Dropdown"
                        data-ouia-safe="true"
                      >
                        <button
                          aria-expanded="false"
                          aria-haspopup="true"
                          class="pf-c-dropdown__toggle dropdown-toggle"
                          data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                          data-ouia-component-type="PF4/DropdownToggle"
                          data-ouia-safe="true"
                          id="pf-dropdown-toggle-id-1"
                          type="button"
                        >
                          <span
                            class="pf-c-dropdown__toggle-text"
                          >
                            Actions
                          </span>
                          <span
                            class="pf-c-dropdown__toggle-icon"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 320 512"
                              width="1em"
                            >
                              <path
                                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>,
                    }
                  }
                  toggleVariant="default"
                >
                  <button
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="pf-c-dropdown__toggle dropdown-toggle"
                    data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                    data-ouia-component-type="PF4/DropdownToggle"
                    data-ouia-safe={true}
                    disabled={false}
                    id="pf-dropdown-toggle-id-1"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    type="button"
                  >
                    <span
                      className="pf-c-dropdown__toggle-text"
                    >
                      Actions
                    </span>
                    <span
                      className="pf-c-dropdown__toggle-icon"
                    >
                      <CaretDownIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 320 512"
                          width="1em"
                        >
                          <path
                            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                          />
                        </svg>
                      </CaretDownIcon>
                    </span>
                  </button>
                </Toggle>
              </DropdownToggle>
            </div>
          </DropdownWithContext>
        </Dropdown>
      </BaseDropdown>
    </Router>
  </BrowserRouter>
</Provider>
`;
